lib/src/entrypoint.dart

```diff
@@ -797,26 +800,41 @@ To update `$lockFilePath` run `$topLevelProgram pub get`$suffix without
     return locked != null && dep.allows(locked);
   }
 
-  /// Determines whether all of the packages in the lockfile are already
-  /// installed and available.
-  bool _arePackagesAvailable() {
-    return lockFile.packages.values.every((package) {
-      if (package.source is UnknownSource) return false;
-
-      // We only care about cached sources. Uncached sources aren't "installed".
-      // If one of those is missing, we want to show the user the file not
-      // found error later since installing won't accomplish anything.
-      var source = package.source;
-      if (source is! CachedSource) return true;
-
-      // Get the directory.
-      var dir = cache.getDirectory(package, relativeFrom: '.');
-      // See if the directory is there and looks like a package.
-      return fileExists(p.join(dir, 'pubspec.yaml'));
-    });
+  /// Determines whether all of the packages in the [packageConfig] are
+  /// older than [packageConfigStat].
+  ///
+  /// The [packageConfigStat] is passed as an optimization to avoid statting the
+  /// same file again.
```

Comment:
Before it used to do:
```
  /// Determines whether all of the packages in the lockfile are already
  /// installed and available.
```

that it doesn't anymore, because you ignore missing pubspecs